apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts

```diff
@@ -145,6 +147,20 @@ export class ParseEventRequest {
       };
     }
 
+    if (template.status === WorkflowStatusEnum.INACTIVE && !this.isGracefulTriggerAllowed(command.payload.__source)) {
+      return {
+        acknowledged: true,
+        status: TriggerEventStatusEnum.NOT_ACTIVE,
+      };
+    }
+
+    if (template.status === WorkflowStatusEnum.ERROR && !this.isGracefulTriggerAllowed(command.payload.__source)) {
+      return {
+        acknowledged: true,
+        status: TriggerEventStatusEnum.NOT_ACTIVE,
```

Comment:
Can we report a better status here? And potentially also pass the actual errors in the response? I would also actually throw a 4XX error here instead of returning 200

Reply:
The status code depends on how we handled the trigger endpoint. Returning 4XX can be a breaking change for some customers who don't handle errors when doing `novu.trigger`. So, for now, we should return whatever the API was returning.